### ISSUE TITLE
Quoted identifiers are not accepted any more (+ too many keywords?)

### ISSUE THREAD

**Issue by puetsch** [sqlfluff/sqlfluff#2654]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
select a as "cost"
from b

```

### Expected Behaviour

Query passes in dialects `redshift` and `postgres`. In previous SQLFluff versions, using keywords as identifiers was fine when they were quoted.

I'm also surprised that `cost` would me a keyword at all. (It's listed as _non-reserved_ in the [doc](https://www.postgresql.org/docs/14/sql-keywords-appendix.html).) Redshift also lists `id` as reserved keyword which seems odd as well.

### Observed Behaviour

I'm getting
> L059 | Unnecessary quoted identifier "cost".

When I remove the quotes, I get
> L029 | Keywords should not be used as identifiers.

### How to reproduce

`sqlfluff lint ../test3.sql`

### Dialect

`redshift` and `postgres`

### Version

sqlfluff, version 0.10.0
Python 3.8

### Configuration

```
[sqlfluff]
verbose = 0
nocolor = False
dialect = postgres
templater = jinja
rules = None
exclude_rules = L009,L011,L019,L031,L034
recurse = 0
output_line_length = 160
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 160
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = consistent

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
This exact situation [was discussed recently on Slack recently](https://sqlfluff.slack.com/archives/C01GX4L213J/p1644526271720919).

The two rules are measuring subtly different things.

> L059 | Unnecessary quoted identifier "cost".

This is saying you don't NEED quotes because `cost` is not a RESERVED keyword.

It is however an UNRESERVED keyword, so best practice it not to use it as an identifier at all, even though, as an unreserved keyword you CAN use it as an identifier if you wish. So it's not an error, but it is something that L029 looks for:

> L029 | Keywords should not be used as identifiers.

So how to resolve?

Well it depends if you find these rules useful or not? And then either obey the rules (and don't use keywords as identifiers at all whether quoted or unquoted) or configure SQLFluff to allow this.

Personally I think L059 is useful and you don't need the quotes her so think that is right.

L029 is more interesting because in a lot of cases you want to use Unreserved keywords and, by the rules of SQL you can (even if it's not best practice). So you've a few choices:

- Disable rule L029 as it's not a rule you agree with. That's OK if that's how you feel - I disable this one myself as just don't find it that useful to be honest (the whole point of them being unreserved, or quoted for reserved is that they CAN be used as identifiers if you want to - though I agree best practice would be not to use them to avoid confusion).
- Add `COST` to the new `ignore_words param` for rule L029, so that particular underserved word never flags in any of your queries, but you can keep on using rule L029 for other unreserved words.
- Add `--noqa: L029` to the end of the offending line too ignore L029 for this line only.

SQLFluff is working as intended here so going to close this issue as non-issue.

**Comment by tunetheweb** [comment]
Oh and btw the reason it's flagged now is that L059 was only added in 0.9.2. So feel free to turn that rule off if it's not helpful, to return to the old functionality.

**Comment by puetsch** [comment]
Thanks for the clarification. That makes sense to me.
